drivers/input/input_ft5336.c

```diff
@@ -50,6 +51,7 @@ LOG_MODULE_REGISTER(ft5336, CONFIG_INPUT_LOG_LEVEL);
 
 /** FT5336 configuration (DT). */
 struct ft5336_config {
+	struct input_touchscreen_common_config common;
```

Comment:
Should probably add a comment here noting that this structure member needs to be first

Reply:
oh good point, for the keyboard stuff I added this:

https://github.com/zephyrproject-rtos/zephyr/blob/main/include/zephyr/input/input_kbd_matrix.h#L264-L268
https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/input/input_gpio_kbd_matrix.c#L42-L43

I'd rather have this than a comment, either can be forgotten in a new driver but at list this would fire if the struct get changed.

Reply:
`INPUT_TOUCH_STRUCT_CHECK` has been added and used in this driver